### ISSUE TITLE
acme provider does not reuse registered account

### ISSUE THREAD

**Issue by zyclonite** [containous/traefik#3650]
### Do you want to request a *feature* or report a *bug*?

Bug

### What did you do?

* starting traefik
* creates acme user account and requests certificates
* configure a new cert or have one installed which expires soon (or expired)
* restart traefik
* creates acme account **again** and requests new certificates

### What did you expect to see?

reuse of the already registered account

### What did you see instead?

```
DEBU[2018-07-20T15:52:34+02:00] Building ACME client...                      
DEBU[2018-07-20T15:52:34+02:00] https://acme-staging-v02.api.letsencrypt.org/directory 
INFO[2018-07-20T15:52:34+02:00] Register...                                  
INFO[2018-07-20T15:52:34+02:00] legolog: [INFO] acme: Registering account for test@mydomain.com 
```

### Output of `traefik version`: (_What version of Traefik are you using?_)

```
Version:      d8f69700e6821465d818b769816710aa5f20908c
Codename:     cheddar
Go version:   go1.10.3
Built:        2018-07-20_02:18:47PM
OS/Arch:      linux/amd64
```
and the same behavior in the latest released version **1.6.5**

### What is your environment & configuration (arguments, toml, provider, platform, ...)?

```toml
[acme]
  email = "test@mydomain.com"
  storage = "/etc/traefik/acme.json"
  entryPoint = "https"
  acmeLogging = true
  onDemand = false
  caServer = "https://acme-staging-v02.api.letsencrypt.org/directory"
```

### Possible root cause

https://github.com/containous/traefik/blob/d8f69700e6821465d818b769816710aa5f20908c/provider/acme/provider.go#L133

here the CAServer is compared against the account uri, which will never match on prefix...
```
https://acme-staging-v02.api.letsencrypt.org/directory != https://acme-staging-v02.api.letsencrypt.org/acme/acct/1234567
```
maybe parsing the URI and matching the hostname would be fine

**Comment by dduportal** [comment]
Hi @zyclonite !

Thanks for your interest in this project.

I was not able to reproduce your issue with the informations you've put here: The toml extract is not complete. I infered some missing attributes, but was not able to reproduce your scenario (Creating a certificate, it create my account, I revoke the cert, restart traefik: the account is not re-created).

I'm going to ask for help about the potential code issue you are mentioning (ping @mmatur : do you know this part?), but it looks like there is something else. Could you add more reproduction information please?

I am particularly interested in the acme.json file: is it located on a data volume? Is it available when you restart? If it is really a bug, fixing it would need reproduction to be possible.

**Comment by zyclonite** [comment]
it's a bit hard to have you reproduce exactly what i did, because i used my own internal acme implementation and configured a certificate expiration of 10 minutes. so traefik would try to renew the certificate on restart.

tomorrow when i am back at work, i'll try to find another way to reproduce that behaviour

i'll test as well the difference before and after that change https://github.com/containous/traefik/commit/3f5772c62af70fafcfe1e94894ee940858e81b7a#diff-07d134a1f5f5ef1572dd887f95bc06b3R118

**Comment by dduportal** [comment]
@zyclonite fair enough, when I say "reproduce", I mean "enough technical information without leaking personal informations as private keys, etc.".

But in this case it would help us to have more details like:

- What is the (docker?) command you used with the arguments passed to traefik ? 
- The docker's detail about where do you write (is the acme.json file in a bind-moutned data volume ?
- The whole traefik.json content (where you replace domain name by foo.bar), etc.

**Comment by zyclonite** [comment]
sure no problem, i'll create a PR for that fix this afternoon

**Comment by dduportal** [comment]
Hello @zyclonite , thank you very much for the effort you've put on this topic!

I am now closing the issue since your PR was merged into the `v1.7` branch.

Feel free to re-open if the upcoming `1.7.0-rc3` release of Træfik is not fixing your challenge :)
